oauth2/src/main/java/com/okta/spring/boot/oauth/env/OktaOAuth2PropertiesMappingEnvironmentPostProcessor.java

```diff
@@ -234,4 +233,48 @@ public Object getProperty(String name) {
             return null;
         }
     }
+
+    /**
+     * PropertySource that resolves <code>okta.oauth2.issuer-with-path</code> property from
+     * <code>okta.oauth2.issuer</code>. Okta has two types of authorization servers (issuers), and the URL format is
+     * slightly different between them, this PropertySource will dynamically return the base URL for each type at
+     * property lookup time.
+     */
+    private static class OktaIssuerWithPathPropertySource extends PropertySource<String> {
+
+        private final Environment environment;
+
+        private OktaIssuerWithPathPropertySource(Environment environment) {
+            super("okta-issuer-url-resolving-source");
+            this.environment = environment;
+        }
+
+        @Override
+        public Object getProperty(@NotNull String key) {
+
+            // only support this key
+            if (OKTA_OAUTH_ISSUER_WITH_PATH.equals(key)) {
+                // issuer could be null
+                return Optional.ofNullable(environment.getProperty(OKTA_OAUTH_ISSUER))
+                    .map(issuer -> {
+                        // Check if URL is a Okta Org Authorization Server
+                        if (!issuer.contains("/oauth2")) {
+                            // for the Okta Org Authorization Server, /oauth2 needs to be appended
+                            issuer += "/oauth2";
+                        }
+                        // for all other issuers leave the property as is
+                        return issuer;
+                    })
+                    // otherwise return null
+                    .orElse(null);
+            }
```

Comment:
```suggestion
            String issuer = environment.getProperty(OKTA_OAUTH_ISSUER);

            // only support this key
            if (OKTA_OAUTH_ISSUER_WITH_PATH.equals(key) && issuer != null) {
                if (!issuer.contains("/oauth2")) {
                    // for the Okta Org Authorization Server, /oauth2 needs to be appended
                    issuer += "/oauth2";
                }
                // for all other issuers leave the property as is
                return issuer;
            }
```
Feel free to leave it as it is, but in my opinion, this looks better

Reply:
@VitaliiTytarenko-okta I agree your block looks better, but after trying it out I realized it didn't work because creates an infinite loop because `environment.getProperty("okta.oauth2.issuer")` may contain the other nested properties.
So, it's only safe to lookup that property after asserting the current key is `OKTA_OAUTH_ISSUER_WITH_PATH`)

I did update the comment though!